db/schema.rb

```diff
@@ -113,9 +121,14 @@
       SELECT e.id,
       e.submission_version,
       e.event_on,
-      (a.application ->> 'service_type'::text) AS service_type
-     FROM (all_events e
+      (a.application ->> 'service_type'::text) AS service_key,
+          CASE
+              WHEN ((a.application ->> 'service_type'::text) = 'custom'::text) THEN ((a.application ->> 'custom_service_name'::text))::character varying
+              ELSE s.translation
+          END AS service
+     FROM ((all_events e
        JOIN application_version a ON (((a.application_id = e.id) AND (a.version = e.submission_version))))
+       JOIN service_translations s ON (((a.application ->> 'service_type'::text) = (s.key)::text)))
```

Comment:
this will not show records where a translation doesn't exist....

```
      SELECT e.id,
      e.submission_version,
      e.event_on,
      (a.application ->> 'service_type'::text) AS service_key,
      COALESCE(a.application ->> 'custom_service_name', COALESCE(s.translation, a.application ->> 'service_type'::text)) AS service
     FROM ((all_events e
       JOIN application_version a ON (((a.application_id = e.id) AND (a.version = e.submission_version))))
       LEFT JOIN service_translations s ON (((a.application ->> 'service_type'::text) = (s.key)::text)))
```
> NOTE: This assumes custom_service_name is `NULL` when type isn't custom

Reply:
I think the assumption in the note is ok considering those are the soft rules of the provider application 
Thanks for the suggestion!

Reply:
updated